lib/Infer/AbstractInterpreter.cpp

```diff
@@ -968,7 +967,83 @@ namespace souper {
     }
     return Result;
   }
-}
 #undef RB0
 #undef RB1
 #undef RB2
+
+#define MDB0 findMustDemandedBits(I->Ops[0])
+#define MDB1 findMustDemandedBits(I->Ops[1])
+#define MDB2 findMustDemandedBits(I->Ops[2])
+#define IVARS Uses.independentVars(I->Ops[0], I->Ops[1])
+
+  std::unordered_map<Inst *, llvm::APInt>
+          MustDemandedBitsAnalysis::findMustDemandedBits(souper::Inst* I) {
+    if (Cache.find(I) != Cache.end()) {
+      return Cache[I];
+    }
+    std::unordered_map<Inst *, llvm::APInt> Result;
+    switch (I->K) {
+      case Inst::Var:
+        Result[I] = llvm::APInt(I->Width, -1);
```

Comment:
as I commented on the original PR, this almost certainly isn't correct -- make sure you've addressed all comments on the original PR